homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,225 @@
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+
+            name = user_input.get(CONF_NAME, CONF_HOST)
+            self.context["title_placeholders"] = {"name": name}
+            if is_ip_address(user_input[CONF_HOST]):
+                unique_id = await self.hass.async_add_executor_job(
+                    functools.partial(get_mac_address, ip=user_input[CONF_HOST])
+                )
+            else:
+                unique_id = await self.hass.async_add_executor_job(
+                    functools.partial(get_mac_address, hostname=user_input[CONF_HOST])
+                )
+
+            # check exist
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            # save for pairing
+            self.user_input = user_input
+
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        errors = {}
+
+        host = self.user_input.get(CONF_HOST)
+        _LOGGER.debug("LG webOS TV %s needs to be paired", host)
+
+        if user_input is not None:
+            try:
+                client = await async_control_connect(self.hass, host, None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+            if "base" not in errors:
+                return await self.async_step_register(self.user_input, client)
+
+        return self.async_show_form(
+            step_id="pairing",
+            errors=errors,
+        )
+
+    async def async_step_register(self, user_input, client=None):
+        """Register entity."""
+        if client.is_registered():
+            name = user_input.get(CONF_NAME, CONF_HOST)
+            user_input["model"] = client.software_info.get("model_name")
+            maj_v = client.software_info.get("major_ver")
+            min_v = client.software_info.get("minor_ver")
+            user_input[CONF_CLIENT_SECRET] = client.client_key
+            user_input["sw_version"] = f"{maj_v}.{min_v}"
+
+            return self.async_create_entry(title=name, data=user_input)
+
+        return await self.async_step_user()
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Webostv device."""
+        user_input = {
+            CONF_HOST: urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname,
+            CONF_NAME: discovery_info["friendlyName"],
+        }
+        self.context["title_placeholders"] = {"name": user_input[CONF_NAME]}
```

Comment:
Does ssdp have the MAC address?  If so can you set the unique id early and abort if it's already configure to avoid making a connecting to the device?

Reply:
Unfortunately the Mac is not going up during UPnP discovery
![image](https://user-images.githubusercontent.com/1258123/118364794-3f81e300-b59a-11eb-9dc6-8b7a600eac22.png)

Reply:
Is the last part of the UDN the Mac ?